pymatgen/entries/compatibility.py

```diff
@@ -648,3 +721,277 @@ def __init__(self, compat_type="Advanced", correct_peroxide=True,
+        self.coeff_mat = []
+        self.exp_uncer = []
+
+        self.mpids = []
+        for cmpd_info in self.exp_compounds:
+            name = cmpd_info['formula']
+            warnings = cmpd_info['warnings']
+
+            if allow_polyanions:
+                warnings.pop('polyanion', None)
+            if allow_large_errors:
+                warnings.pop('large_uncertainty', None)
+            if allow_unstable:
+                warnings.pop('unstable', None)
+
+            if name in self.calc_compounds and not warnings:
+                
+                comp = Composition(name)
+                elems = list(comp.as_dict())
+                
+                compound = self.calc_compounds[name]
+                
+                reactants = []
+                for elem in elems:
+                    try:
+                        reactants.append(self.calc_compounds[elem])
+                    except KeyError:
+                        raise ValueError('Computed entries missing ' + elem)
+                
+                rxn = ComputedReaction(reactants, [compound])
+                rxn.normalize_to(comp)
+                energy = rxn.calculated_reaction_energy
+                
+                if compound.data['oxide_type'] == 'oxide':
+                    coeff = [comp['O'], 0, 0]
+                    self.oxides.append(name)
+                elif compound.data['oxide_type'] == 'peroxide':
+                    coeff = [0, comp['O'], 0]
+                    self.peroxides.append(name)
+                elif compound.data['oxide_type'] == 'superoxide':
+                    coeff = [0, 0, comp['O']]
+                    self.superoxides.append(name)
+                else:
+                    coeff = [0, 0, 0]
+                coeff += [comp[elem] for elem in self.species[3:]]
+                
+                self.names.append(name)
+                self.diffs.append((cmpd_info['exp energy'] - energy)/comp.num_atoms)
+                self.coeff_mat.append([i/comp.num_atoms for i in coeff])
+                self.exp_uncer.append((cmpd_info['uncertainty'])/comp.num_atoms)
+                            
+                self.mpids.append(compound.entry_id)   
+
+        #for any exp entries with no uncertainty value, assign average uncertainty value
+        sigma = np.array(self.exp_uncer)
+        sigma[sigma == 0] = np.nan
+        mean_uncer = np.nanmean(sigma)
+        sigma = np.where(np.isnan(sigma), mean_uncer, sigma)
+
+        f = lambda x, *m: np.dot(x, m)
+
+        popt, pcov = curve_fit(f, self.coeff_mat, self.diffs, p0=np.ones(21), sigma=sigma, absolute_sigma=True)
+        self.corrections = popt.tolist()
+        self.corrections_std_error = np.sqrt(np.diag(pcov)).tolist()
+        for i in range(len(self.species)):
+            self.corrections_dict[self.species[i]] = (self.corrections[i], self.corrections_std_error[i])
+        return self.corrections_dict
+
+
+    def graph_residual_error(self):
```

Comment:
Switch to plotly plots so that we can have them interactive and annotated for more information.